mysys/checksum.c

```diff
@@ -28,14 +70,13 @@
       length    length of the block
 */
 
-ha_checksum my_checksum(ha_checksum crc, const uchar *pos, size_t length)
+inline __attribute__((nonnull, always_inline)) ha_checksum
+my_checksum(ha_checksum crc, const uchar *pos, size_t length)
 {
-#ifdef HAVE_CRC32_VPMSUM
-  extern unsigned int crc32ieee_vpmsum(unsigned int crc, const unsigned char *p,
-                                    unsigned long len);
-  crc= (ha_checksum) crc32ieee_vpmsum((uint) crc, pos, (uint) length);
+#if defined(HAVE_CRC32_VPMSUM)
+  crc= crc32ieee_vpmsum(crc, pos, length);
 #else
-  crc= (ha_checksum) crc32((uint)crc, pos, (uint) length);
+  crc= my_crc32((uint) crc, pos, (uint) length);
 #endif
   DBUG_PRINT("info", ("crc: %lu", (ulong) crc));
   return crc;
```

Comment:
I think that `my_checksum` should be a function pointer that points to one of the implementations directly.

Reply:
commented above on my_checksum indirection.

Reply:
I do not use DBUG for anything but DBUG_EXECUTE_IF , or DBUG_ASSERT, so yes, *I* do not mind removing the print really.

Reply:
As commented function pointer is now moved to my_sys.h and renamed to my_checksum so we no more maintain additional indirection.